### ISSUE TITLE
Progen2 for a series of sequences

### ISSUE THREAD

**Issue by Lsz-20** [hugohrban/ProGen2-finetuning#11]
I try to input a series sequences,but it seems not work well now.
How should I modify the following section:
target = torch.tensor(tokenizer.encode(tokens).ids).to(device)
tokens now is a list of Encoding.
Thanks

**Comment by hugohrban** [comment]
Hi, which section are you reffering to? 
tokenizer.encode takes as input a string. 
If tokens is a list of Encoding's, you can't encode an Encoding. You can refer to documentation of the tokenizers library [here](https://huggingface.co/docs/tokenizers/api/tokenizer#tokenizers.Tokenizer.encode)

**Comment by Lsz-20** [comment]
Currently my tokens is a list with multiple string loaded,if i want to run this code, i need to traverse this list to encode this sequences right?
And another question, to achieve padding and batch encode，i try the following：
  "tokenizer.enable_padding()" 
and 
"torch.tensor(tokenizer.encode_batch(tokens).ids).to(device)" replace "torch.tensor(tokenizer.encode(tokens).ids).to(device)"
And I got the following result：
tensor([[ *, *, *, *, *, *],
        [ *, *, *, *, *, *],
        [ *, *, *, *, *, *]])
Is this result correct

**Comment by Lsz-20** [comment]
For the source code, the target and logits's shape are:[lenth] [lenth, channel],if use batch_encode it change to [batch, lenth] [batch,lenth,channel],its right for cross_entropy？
